server/routes/dates.js

```diff
@@ -0,0 +1,111 @@
+const express = require('express')
+const moment = require('moment')
+
+const cardData = require('../db/cardData')
+const graph = require('../db/graph')
+const activities = require('../db/activities')
+
+const router = express.Router()
+
+module.exports = router
+
+// router.get('/cards', (req, res) => {
+//   cardData.getCards()
+// })
+
+router.post('/', (req, res) => {
+  // req.body required to look like :
+  // {userId: 1, records:[{activityId:1, rating:1, log:'blah'},{more cards}]}
+  const {userId, records} = req.body
+
+  // if (isRecordOkay) {
+  cardData.addDate({user_id: userId})
+    .then(dateId => {
+      for (let rec of records) {
+        rec.date_id = dateId[0]
+        rec.activity_id = rec.activityId
+        delete rec.activityId
+      }
+
+      cardData.addRecords(records)
+        .then(() => {
+          // res.json(records)
+          res.json({Okay: true})
+        })
+        .catch((err) => res.json({Okay: false, error: err.message}))
+    })
+    .catch((err) => res.json({Okay: false, error: err.message}))
+
+  // }
+  // else {
+  //   res.json({Okay: false, error: 'Data format is not correct'})
+  // }
+})
+
+router.get('/graph/:userId/:endDate', (req, res) => {
+  const userId = Number(req.params.userId)
+  let endDate = req.params.endDate
+  // let endDate = '2018-12-14'
+  endDate += ' 23:59:59'
+  const period = 'week'
+
+  let date = moment(endDate)
+  let startDate = date.add(-1, period).format('YYYY-MM-DD')
+  let chartData = {}
```

Comment:
Probably use `const` for most things unless it's actually generating an error. It can protect you against subtle bugs, where you reassign something you didn't mean to.